### PR TITLE
Back button fixes

### PR DESCRIPTION
**Pull request by srallen**
Staging branch URL: https://back-button-fixes.pfe-preview.zooniverse.org/

Fixes a few of the issues listed in #4570:
- if no task help, Back and Next/Done very close to task
  - Reworked bottom margins between components
- The Back button isn't disabled on the first task of a workflow, even though it can't be used. It also responds to hover on the first task, and displays the warning message.
  - Back button is now hidden on first task
- Line heights don't match for the task navigation buttons. (Done & Talk)
  - Added explicit line height
- "Going back..." warning remains even if not hover on Back button, disappears on hover 
  - Now using separate explicit handlers for mouseenter and focus vs mouseleave and blur.

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/classifier/components/TaskNavButtons/components/TalkLink/TalkLink.jsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export const StyledTalkLink = styled(Link)`
 export const StyledDisabledTalkPlaceholder = styled.span`
   ${commonStyles}
   opacity: 0.5;
-  pointer-events: 'none';
+  pointer-events: none;
```
Comment (eatyourgreens) [inline app/classifier/components/TaskNavButtons/components/TalkLink/TalkLink.jsx]: Nice solution to disabling links. Does it disable them from the keyboard too? If not, consider adding `tabindex="-1"` when the link is disabled.

**Comment by eatyourgreens** [comment]
The Back button is a different height from the other buttons on Scribes of the Cairo Geniza.
Screenshot showing Back button about half the height of Done and Done & Talk.

Other than that, this looks good.

**Comment by eatyourgreens** [comment]
Approving this so as not to hold up merging, since it solves a few issues that have been remarked upon in Talk. In particular, the Back button showing up for the first task of any workflow.

**Comment by srallen** [comment]
@eatyourgreens Thanks. I didn't include a fix for the button size when Done & Talk wraps because I wanted to get @beckyrother input and she was quite busy yesterday. I'll get it in another pass.

**Comment by eatyourgreens** [comment]
@srallen one other small thing that a couple of people have pointed out on Talk: when Done & Talk is disabled, and rendered as a span, it still has `cursor: pointer` as a style.

**Pull request merged**